internal/rootcoord/root_coord.go

```diff
@@ -2785,10 +2785,8 @@ func (c *Core) CheckHealth(ctx context.Context, in *milvuspb.CheckHealthRequest)
 
 	mu := &sync.Mutex{}
 	group, ctx := errgroup.WithContext(ctx)
-	errReasons := make([]string, 0, len(c.proxyClientManager.proxyClient))
-
-	c.proxyClientManager.lock.RLock()
-	for nodeID, proxyClient := range c.proxyClientManager.proxyClient {
+	errReasons := make([]string, 0, c.proxyClientManager.GetProxyCount())
+	for nodeID, proxyClient := range c.proxyClientManager.GetProxyClients() {
```

Comment:
Returning the underlying map object directly here does not guarantee concurrent safety.

Reply:
Agreed, we have use sync map or copy